geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/protobufs/TupleProtoBuf.scala

```diff
@@ -50,6 +74,17 @@ trait TupleProtoBuf {
         multibandTileProtoBufCodec.decode(message.tiles.get))
   }
 
+  implicit def tupleSpatialKeyImageProtoBufCodec =
+    new ProtoBufCodec[(SpatialKey, Array[Byte]), ProtoTuple] {
+    def encode(tuple: (SpatialKey, Array[Byte])): ProtoTuple =
+      ProtoTuple(
+        spatialKey = Some(spatialKeyProtoBufCodec.encode(tuple._1)),
+        imageBytes = ByteString.copyFrom(tuple._2))
```

Comment:
Let me actually throw it into the compiler to double check. I see that I missed a type parameter in my mockup. If it works it would remove a lot of the code introduced in this PR, so it should really be here.